services/database-analyzer.js

```diff
@@ -1,265 +1,38 @@
-            //         handle it automatically without necessary declaration.
-            if (!(nameColumn === 'id' && type === 'INTEGER' && columnInfo.primaryKey)) {
-              const field = {
-                name: _.camelCase(nameColumn),
-                nameColumn,
-                type,
-                primaryKey: columnInfo.primaryKey,
-                defaultValue: columnInfo.defaultValue,
-              };
-
-              fields.push(field);
-            }
-          }
-        });
-
-        const options = {
-          underscored: isUnderscored(fields),
-          timestamps: hasTimestamps(fields),
-          hasIdColumn: hasIdColumn(fields, primaryKeys),
-          hasPrimaryKeys: !_.isEmpty(primaryKeys),
-        };
-
-        return {
-          fields,
-          references,
-          primaryKeys,
-          options,
-        };
-      });
-  }
-
-  async function sequelizeTableAnalyzer() {
-    const schema = {};
-
-    queryInterface = databaseConnection.getQueryInterface();
-    tableForeignKeysAnalyzer = new TableForeignKeysAnalyzer(databaseConnection, config);
-    columnTypeGetter = new ColumnTypeGetter(databaseConnection, config.dbSchema || 'public', allowWarning);
-
-    if (config.dbSchema) {
-      const schemaExists = await queryInterface.sequelize
-        .query(
-          'SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = ?;',
-          { type: queryInterface.sequelize.QueryTypes.SELECT, replacements: [config.dbSchema] },
-        )
-        .then(result => !!result.length);
-
-      if (!schemaExists) {
-        logger.error('This schema does not exists.');
-        return process.exit(1);
-      }
-    }
-
-    // Build the db schema.
-    await P.mapSeries(queryInterface.showAllTables({
-      schema: config.dbSchema,
-    }), async (table) => {
-      // NOTICE: MS SQL returns objects instead of strings.
-      // eslint-disable-next-line no-param-reassign
-      if (typeof table === 'object') { table = table.tableName; }
-
-      schema[table] = await analyzeTable(table);
-    });
+async function reportEmptyDatabase(orm, dialect) {
+  const logs = [`Your database looks empty! Please create some ${orm === 'mongoose' ? 'collections' : 'tables'} before running the command.`];
+  if (orm === 'sequelize') {
+    logs.push('If not, check whether you are using a custom database schema (use in that case the --schema option).');
+  }
+  logger.error(...logs);
+  await eventSender.notifyError('database_empty', 'Your database is empty.', {
+    orm,
+    dialect,
+  });
+  return process.exit(1);
+}
 
-    if (_.isEmpty(schema)) {
-      return reportEmptyDatabase('sequelize', databaseConnection.getDialect());
+function DatabaseAnalyzer(databaseConnection, config, allowWarning) {
+  this.perform = async () => {
+    let analyze;
+    if (config.dbDialect === 'mongodb') {
+      analyze = mongoTableAnalyzer;
```

Comment:
`mongoTableAnalyzer` & `sequelizeTableAnalyzer` are function, yes. 
To avoid the weird look assignation, we may spell them `mongoAnalyze` & `sequelizeAnalyze`?

I understand the point about the inconsistent naming of the two analyzers, one is orm and one is DB. It's not the same level.
But I don't feel agree because they are concrete implementations. They should be named by what they do, at the lower level. I figure that because if we add another NoSql DB tomorrow (let's say Cassandra, CouchBase, Hazelcast or any), we won't be able to name the analyzer. We may also add other ways to analyze SQL, too?

Reply:
all right, then, go for:
`mongoCollectionsAnalyze` & `sequelizeTablesAnalyze`
or
`analyzeMongoCollections` & `analyzeSequelizeTables`

And lets reflect this choice in the file names and `exports`.

👍

Reply:
👍